packages/scheduler/src/ClockTimer.ts

```diff
@@ -0,0 +1,58 @@
+/** @license MIT License (c) copyright 2010-2017 original author or authors */
+
+import { defer, DeferrableTask } from './task'
+import { Clock, Time, Timer } from '@most/types'
+
+/* global setTimeout, clearTimeout */
+
+export default class ClockTimer implements Timer {
+  private _clock: Clock;
+  constructor (clock: Clock) {
+    this._clock = clock
+  }
+
+  now (): Time {
+    return this._clock.now()
+  }
+
+  setTimer <A> (f: () => A, dt: Time): NodeJS.Timeout | Asap<A> {
+    return dt <= 0 ? runAsap(f) : setTimeout(f, dt)
+  }
+
+  clearTimer <A> (t: number | Asap<A>): void {
+    return t instanceof Asap ? t.cancel() : clearTimeout(t)
+  }
+}
+
+class Asap<A> implements DeferrableTask<never, A | undefined> {
```

Comment:
Can you help me understand why this needs to be `A | undefined` rather than just `A`?

Reply:
That's because not all parts of `Asap#run` return `A`. Anyway I removed `| undefined` from `Asap#f`

Reply:
Got it, thanks.  Is it possible at least restrict `f` to be of type `() => A`?

Reply:
Yeah, that's exactly what I did :)

Reply:
👍